sound/soc/sof/intel/hda-codec.c

```diff
@@ -113,8 +114,14 @@ static int hda_codec_probe(struct snd_sof_dev *sdev, int address,
 	if (ret < 0)
 		return ret;
 
-	if ((resp & 0xFFFF0000) == IDISP_VID_INTEL)
+	if ((resp & 0xFFFF0000) == IDISP_VID_INTEL) {
+		if (!hda->i915_codec_enabled) {
+			dev_dbg(sdev->dev,
+				"iDisp hw present but no driver, ignoring\n");
+			return -ENOENT;
```

Comment:
One thing I wondered about is should the device be unregistered here immediately in the error case. But I ended up  not doing that. snd_hdac_ext_bus_device_remove() will clean up this device as well.